etna/analysis/plotters.py

```diff
@@ -1958,3 +1959,116 @@ def update(**kwargs):
         plt.show()
 
     interact(update, **sliders)
+
+
+class ComponentsMode(str, Enum):
+    """Enum for components plotting modes."""
+
+    per_component = "per-component"
+    together = "together"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise NotImplementedError(
+            f"{value} is not a valid {cls.__name__}. Supported modes: {', '.join([repr(m.value) for m in cls])}"
+        )
+
+
+def plot_forecast_decomposition(
+    forecast_ts: "TSDataset",
+    test_ts: Optional["TSDataset"] = None,
+    mode: Union[Literal["per-component"], Literal["together"]] = "per-component",
+    segments: Optional[List[str]] = None,
+    columns_num: int = 1,
+    figsize: Tuple[int, int] = (10, 5),
+    show_grid: bool = False,
+):
+    """
+    Plot of prediction and its components.
+
+    Parameters
+    ----------
+    forecast_ts:
+        forecasted TSDataset with timeseries data, single-forecast mode
+    test_ts:
+        TSDataset with timeseries data
+    mode:
+        Components plotting type
+
+        #. ``per-component`` -- plot each component in separate axes
+
+        #. ``together`` -- plot all the components in the same axis
+
+    segments:
+        segments to plot; if not given plot all the segments
+    columns_num:
+        number of graphics columns; when mode=``per-component`` all plots will be in the single column
+    figsize:
+        size of the figure per subplot with one segment in inches
+    show_grid:
+        whether to show grid for each chart
+
+    Raises
+    ------
+    ValueError:
+        if components aren't present in ``forecast_ts``
+    NotImplementedError:
+        unknown ``mode`` is given
+    """
+    components_mode = ComponentsMode(mode)
+
+    if segments is None:
+        segments = list(forecast_ts.columns.get_level_values("segment").unique())
+
+    column_names = set(forecast_ts.columns.get_level_values("feature"))
+    components = list(match_target_components(column_names))
+
+    if len(components) == 0:
+        raise ValueError("No components were detected in the provided `forecast_ts`.")
+
+    if components_mode == ComponentsMode.together:
+        num_plots = len(segments)
+    else:
+        # separate chart for target/forecast
```

Comment:
Is this comment correct? It looks like we separate chart for all the components + target.

Reply:
Yes, this explains why we add 1 to the total number of components (n_components + separate chart for target). I will rewrite to a more clearer formulation.